sdk/src/main/scala/com/stratio/sparkta/sdk/AggregationTime.scala

```diff
@@ -0,0 +1,86 @@
+/**
+  * Copyright (C) 2015 Stratio (http://stratio.com)
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package com.stratio.sparkta.sdk
+
+import com.github.nscala_time.time.Imports._
+import org.joda.time.Duration
+
+object AggregationTime {
+  val Prefix: Seq[String] = Seq("s","m","h", "d", "y", "second", "minute", "hour", "day", "month", "year")
+
+  def truncateDate(date: DateTime, granularity: String): Long = {
+
+    val duration: Duration = parseDate(date, granularity)
+    roundDateTime(date, duration)
+  }
+
+  def parseDate(date: DateTime, aggregationTime: String): Duration = {
+    val prefix = for {
+      prefix <- Prefix
+      if(aggregationTime.contains(prefix))
```

Comment:
You should change this line replacing it by a match using a regex (Change the Prefix property with regex).